docs/sql.rst

```diff
@@ -0,0 +1,690 @@
+    from hazelcast.config import Config
+    config = Config()
+    config.compact_serializers = [AddressSerializer()]
+    conn = connect(config)
+
+Once the connection is created, you can create a :class:`hazelcast.db.Cursor` object from it
+to execute queries. This is explained in the next section.
+Finally, you can close the `Connection` object to release its resources if you are done with it.
+
+.. code:: python
+
+    conn.close()
+
+You can use a `with` statement to automatically close a `Connection`.
+
+.. code:: python
+
+    from hazelcast.db import connect
+    with connect() as conn:
+        # use conn in this block
+    # conn is automatically closed here
+
+Cursors
+~~~~~~~
+
+The first step of executing a query is, getting a :class:`hazelcast.db.Cursor` from the connection.
+
+.. code:: python
+
+    cursor = conn.cursor()
+
+Then, you can execute a SQL query using the :meth:`hazelcast.db.Cursor.execute` method.
+You can use this method to run all kinds of queries.
+
+.. code:: python
+
+    cursor.execute("SELECT * FROM stocks ORDER BY price")
+
+Use the question mark (`?)` as a placeholder if you are passing arguments
+in the query. The actual arguments should be passed in a tuple.
+
+.. code:: python
+
+    cursor.execute("SELECT * FROM stocks WHERE price > ? ORDER BY price", (50,))
+
+:meth:`hazelcast.db.Cursor.executemany` is also available, which enables running the same query
+with different kinds of value sets. This method should only be used my mutating queries, such as `INSERT`.
+
+.. code:: python
+
+    data = [
+        (1, "2006-03-28", "BUY", "IBM", 1000, 45.0),
+        (2, "2006-04-05", "BUY", "MSFT", 1000, 72.0),
+        (3, "2006-04-06", "SELL", "IBM", 500, 53.0),
+    ]
+    cursor.executemany("INSERT INTO stocks VALUES(?, CAST(? AS DATE), ?, ?, ?, ?)", data)
+
+**Mutating Queries**
+
+Mutating queries such as `UPDATE`, `DELETE` and `INSERT` updates, deletes
+data or adds new rows. You can use `execue` or `executemany` for those
+queries.
+
+.. code:: python
+
+    cursor.execute("INSERT INTO stocks(__key, price) VALUES(10, 40)")
+
+
+**Row Returning Queries**
+
+Queries such as `SELECT` and `SHOW` return rows. Once you run `execute`
+with the query, call one of :meth:`hazelcast.db.Cursor.fetchone`,
+:meth:`hazelcast.db.Cursor.fetchmany` or :meth:`hazelcast.db.Cursor.fetchall`
+to get one, some or all rows in the result. The rows are of the
+:class:`hazelcast.db.SqlRow` type. Note that, `fetchall` should only be used
+for small, finite set of rows.
+
+.. code:: python
+
+    cursor.execute("SELECT * FROM stocks")
+    one_row = cursor.fetchone()
+    three_more_rows = cursor.fetchmany(3)
+    rest_of_rows = cursor.fetchall()
+
+Alternatively, you can iterate on the cursor itself.
+
+.. code:: python
+
+    cursor.execute("SELECT * FROM stocks")
+    for row in cursor:
+        # handle the row
+
+You can access columns in a:class:`hazelcast.db.SqlRow` by using the subscription
```

Comment:
`hazelcast.db` imports `SqlRow` so it can also be imported from there. It's probably better to stick to `hazelcast.db` module in this section of the documentation.

Reply:
Fixed the typo at: https://github.com/hazelcast/hazelcast-python-client/pull/626/commits/a3d30fc80ea4a108bd80ff0ac6af4e383e93e423

Reply:
But, sphinx is not able to resolve it, and does not add the link to the SqlRow API doc.

Reply:
That's strange: https://github.com/hazelcast/hazelcast-python-client/pull/626/commits/c8680b8b8235c89cfff4276c07a5d5840b1bf404